### PR TITLE
Smart Wallets: Update Chain Export

### PR DESCRIPTION
**Pull request by devlyn37**
## Description

This PR:
- removes the `SmartWalletChain` export from `@crossmint/client-sdk-smart-wallet`, and replaces it with the chains from the core sdk package, renamed from `EVMChainsIncludingTestnet` to `Chain`.
- renames the chain export from `@crossmint/client-sdk-smart-wallet-web3auth-adapter`, which is already doing what I mention above, from `Blockchain` to `Chain`.

These are breaking changes, but I only want to do a patch version bump, the SDK shipped less than 24 hrs ago and is in EAP mode so I'm comfortable breaking semantic versioning rules.

## Test plan

Updated the Demo to use the new export and TS compiled.

### CODE REVIEW

File: packages/client/wallets/smart-wallet/src/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export { blockchainToChainId } from "@crossmint/common-sdk-base";
+export { blockchainToChainId, EVMBlockchainIncludingTestnet as Chain } from "@crossmint/common-sdk-base";
```
Comment (devlyn37) [inline packages/client/wallets/smart-wallet/src/index.ts:1]: I think `Chain` is better than `Blockchain` here. fwiw it's how we've named things in `crossbit-main` to

File: apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createMetamaskWallet.ts [hunk 2]
```diff
@@ -24,13 +23,9 @@ export const createMetamaskAAWallet = async (isProd: boolean) => {
...
               });
 
-        return await xm.getOrCreateWallet(
-            userIdentifier,
-            isProd ? SmartWalletChain.POLYGON : SmartWalletChain.POLYGON_AMOY,
-            {
-                signer: window.ethereum as any,
-            }
-        );
+        return await xm.getOrCreateWallet(userIdentifier, isProd ? Chain.POLYGON : Chain.POLYGON_AMOY, {
```
Comment (afeight) [inline apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createMetamaskWallet.ts:26]: nit: `xm` => `smartWalletSDK` or like `smartWallets`

File: apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createMetamaskWallet.ts [hunk 3]
```diff
@@ -24,13 +23,9 @@ export const createMetamaskAAWallet = async (isProd: boolean) => {
...
+            signer: window.ethereum as any,
+        });
```
Comment (afeight) [inline apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createMetamaskWallet.ts:28]: nit: might be a bit too much going on in one line
```suggestion
        const blockchain = isProd ? Chain.POLYGON : Chain.POLYGON_AMOY
        return await xm.getOrCreateWallet(userIdentifier, blockchain, { signer: window.ethereum as any });
```

**Review by afeight — APPROVED** [review]
tentative approval to unblock, but do feel that `Blockchain` is better for our users because: 
1. they're not typically as blockchain-savvy as us
2. code editors will find `blockchain` if you type `chain` or `blockchain`, but not find `chain` if you try to type `blockchain`

File: apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createMetamaskWallet.ts (hunk 2 above)
Comment (devlyn37) [reply apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createMetamaskWallet.ts:26]: Yeah demo is a huge mess atm, jon and I will be cleaning up in separate PRs

File: apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createMetamaskWallet.ts (hunk 3 above)
Comment (devlyn37) [reply apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createMetamaskWallet.ts:28]: agree, but out of scope here. will cleanup demo later w/ jon

**Pull request merged**